sycl/include/sycl/detail/pi.h

```diff
@@ -561,6 +561,19 @@ constexpr pi_memory_scope_capabilities PI_MEMORY_SCOPE_WORK_GROUP = 0x04;
 constexpr pi_memory_scope_capabilities PI_MEMORY_SCOPE_DEVICE = 0x08;
 constexpr pi_memory_scope_capabilities PI_MEMORY_SCOPE_SYSTEM = 0x10;
 
+// CL equivalents are only available for OpenCL version 3.0
+#define PI_DEVICE_ATOMIC_MEMORY_CAPABILITIES 0x1063
```

Comment:
Why do we need this macro here? The only use I see in OpenCL PI plugin. I suggest to define it there directly or even better, use a macro from `CL/cl.h` header

Reply:
Could you not just include `CL/cl_version.h` and use `#ifdef CL_VERSION_3_0` (or similar) to surround where you use those symbols in the `if (devVer >= OCLV::V3_0) {` conditional block for `pi_opencl.cpp`? Would save having to add these redefinitions.

Reply:
> Could you not just use #ifdef CL_VERSION_3_0 (or similar) to surround where you use those symbols in the if (devVer >= OCLV::V3_0) { conditional block for pi_opencl.cpp?

Using such `#ifdef` would make sense if we were aiming to allow building our OpenCL plugin against different versions of OpenCL headers.

If we want to use 3.0 API, then we should probably update the following piece of code:

https://github.com/intel/llvm/blob/ccf205e4a281d47de4196a6b8aec2f9dc0277670/sycl/include/sycl/detail/cl.h#L11-L15

I honestly don't know if 3.0 headers are backwards compatible with previous OpenCL versions so I'm not entirely sure if anything would break.

The bare minimum which I would suggest is to at least move our custom define a level lower, because there seem to be no need to share it across all plugins or users of PI interface.

Reply:
> I honestly don't know if 3.0 headers are backwards compatible with previous OpenCL versions so I'm not entirely sure if anything would break.

The headers are fully backwards compatible.  There is actually only one set of OpenCL headers and the OpenCL version they support is controlled by an preprocessor define, see: https://github.com/KhronosGroup/OpenCL-Headers/#compiling-for-a-specific-opencl-version